sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/utils/HostIpAddressUtils.java

```diff
@@ -48,33 +47,43 @@ private HostIpAddressUtils() {
      * @throws SocketException
      */
     public static String getHostAddress() throws SocketException {
-        try {
-            Enumeration<NetworkInterface> networkInterface = NetworkInterface.getNetworkInterfaces();
-            InetAddress candidateAddress = null;
-            while (networkInterface.hasMoreElements()) {
-                NetworkInterface ni = networkInterface.nextElement();
-                for (Enumeration<InetAddress> inetAdd = ni.getInetAddresses(); inetAdd.hasMoreElements();) {
-                    InetAddress inetAddress = inetAdd.nextElement();
-
-                    // Determine whether it is a loopback address
-                    if (!inetAddress.isLoopbackAddress()) {
-                        // If it is a site-local address, return it directly
-                        if (inetAddress.isSiteLocalAddress()) {
-                            return inetAddress.getHostAddress();
-                        }
-                        if (candidateAddress == null) {
-                            candidateAddress = inetAddress;
-                        }
-                    }
-                }
-            }
+        InetAddress candidateAddress = findNonLoopbackAddress();
+        if (candidateAddress != null) {
+            return candidateAddress.getHostAddress();
+        }
 
-            // If there is no other address outside the loopback, then Inet Address will be obtained directly
-            return candidateAddress == null ? InetAddress.getLocalHost().getHostAddress()
-                    : candidateAddress.getHostAddress();
+        try {
+            return InetAddress.getLocalHost().getHostAddress();
         } catch (UnknownHostException e) {
             LOGGER.warning("get host address error");
         }
+
         return DEFAULT_ADDRESS;
     }
+
+    private static InetAddress findNonLoopbackAddress() throws SocketException {
+        Enumeration<NetworkInterface> networkInterface = NetworkInterface.getNetworkInterfaces();
+        InetAddress candidateAddress = null;
+        while (networkInterface.hasMoreElements()) {
+            NetworkInterface ni = networkInterface.nextElement();
+            for (Enumeration<InetAddress> inetAdd = ni.getInetAddresses(); inetAdd.hasMoreElements(); ) {
+                InetAddress inetAddress = inetAdd.nextElement();
+
+                // Determine whether it is a loopback address
+
```

Comment:
Delete blank line between code and comments.

Reply:
done